docs/integration/java/jira oidc.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Jira OIDC
+description:  Using OIDC protocol as IDP to connect various applications, like Jira
+keywords: [OIDC, Jira, IDP]
+---
+
+Casdoor can use OIDC protocol as IDP to connect various applications. Here we will use Jira as an example to show you how to use OIDC to connect to your applications.
+
+The following are some of the names in the configuration:
+
+`CASDOOR_HOSTNAME`: Domain name or IP where Casdoor server is deployed.
+
+`Jira_HOSTNAME`: Domain name or IP where Jira is deployed.
+
+## Step1. Deploy Casdoor and Jira
+
+Firstly, the [Casdoor](/docs/basic/server-installation) and Jira should be deployed.
+
+After a successful deployment, you need to ensure:
+
+1. Set Jira URL(Plans -> Administration -> System -> General configuration) to `Jira_HOSTNAME`.
+![Jira URL](/img/integration/java/jira-oidc/Jira_HOSTNAME.png)
+2. Casdoor can be logged in and used normally.
+3. My use default pord so my CASDOOR_HOSTNAME = 'http://localhost:8000'
+
+## Step2. Configure Casdoor application
+
+1. Create or use an existing Casdoor application.
+2. Find a redirect url: ![Jira Callback](/img/integration/java/jira-oidc/Jira_CallbackURL.png)
+3. Add a redirect url: ![Jira Application](/img/integration/java/jira-oidc/Jira_application.png)
+4. Add provider you want and supplement other settings.
+
+Not surprisingly, you can get two values ​​on the application settings page: `Client ID` and `Client secret` like the picture above, we will use them in the next step.
+
+Open your favorite browser and visit: **http://`CASDOOR_HOSTNAME`/.well-known/openid-configuration**, you will see the OIDC configure of Casdoor.
+
+## Step3. Configure Jira
+
+1. You should install a app to support OAuth ![Jira App](/img/integration/java/jira-oidc/Jira_install.png)
+2. You should config this app ![Jira Config](/img/integration/java/jira-oidc/Jira_Config.png)
+   - Application you should select Custom OpenId
+   - You can find Client Id and Client Secret in Casdoor application page.
+     - `Token server url`: **http://`CASDOOR_HOSTNAME`/api/login/oauth/access_token**
```

Comment:
Too much indentation. It's not very nice here.

![image](https://user-images.githubusercontent.com/71440988/199975485-24072f56-8420-48a9-a65b-d904ddaebc7c.png)